components/portal/messages/filters.js

```diff
@@ -68,6 +68,31 @@ define(['angular'], function(angular) {
         });
       };
     })
+    .filter('filterMessagesWithInvalidDates', function() {
+      return function(messages) {
+          var thePresent = moment.now();
+          return messages.filter(function(message) {
+            var goDate = moment(message.goLiveDate);
+            var stopDate = moment(message.expireDate);
+            // If neither date is populated, message is valid.
+            if (!goDate.isValid() && !stopDate.isValid()) {
+              return message;
+            }
+            if (goDate.isValid() && stopDate.isValid()) {
+              if (goDate.isBefore(thePresent)
+                && stopDate.isAfter(thePresent)) {
+                return message;
+              }
+            }
+            if (goDate.isValid() && goDate.isBefore(thePresent)) {
+              return message;
+            }
+            if (stopDate.isValid() && stopDate.isAfter(thePresent)) {
+              return message;
```

Comment:
I'm not a fan of multiple return statements, but prior experience indicates I'm outnumbered.

Reply:
`ClojureScript`. If we really want to be experimenting with pure functions, where this is going is writing some of our stuff in `ClojureScript`. And hey then maybe it would be radically more testable.

Reply:
Collapsing at least some of these conditionals might be more literate anyway.  The logic seems to be ORing some ways that the message might pass the filter? So maybe say that, with ye olde `||` operator?

Reply:
> `ClojureScript`. If we really want to be experimenting with pure functions, where this is going is writing some of our stuff in `ClojureScript`. And hey then maybe it would be radically more testable.

There are a number of pure functional languages that compile to JavaScript.
https://github.com/stoeffel/awesome-fp-js#functional-languages-that-compile-to-javascript
https://stateofjs.com/2017/flavors/results

I've heard good things about Elm and Reason, as well as ClojureScript.